aip/client-libraries/4221.md

```diff
@@ -0,0 +1,80 @@
+---
+aip:
+  id: 4221
+  scope: client-libraries
+  state: draft
+  created: 2019-06-17
+permalink: /client-libraries/4221
+redirect_from:
+  - /4221
+---
+
+# Client-side retry
+
+Many APIs have error modes that a client should retry. These error modes vary
+across APIs, leaving users to hand-write retry logic around client libraries
+based on available documentation.
+
+## Guidance
+
+Client libraries **may** provide automatic client-side retry logic. Client
+libraries with automatic client-side retry logic **should** provide mechanisms
+(e.g. client initialization options) for users to configure it.
```

Comment:
This sentence is a bit awkward. Perhaps something like, "...**should** provide a mechanism for users to specify error codes to be retried."

Reply:
I like this better, but we also support supplying "retry delays" (e.g. initial & max backoff, multiplier), so I will include some words on that as well.

Reply:
Done.